gb-cpu/src/microcode/controller.rs

```diff
@@ -156,6 +167,20 @@ impl MicrocodeController {
         let interrupt_enable = self.interrupt_enable;
         interrupt_flag & interrupt_enable != 0
     }
+
+    #[cfg(feature = "registers_logs")]
+    fn log_registers_to_file(&mut self, opcode_logs: &str) -> std::io::Result<()> {
+        let current_dir_path = env::current_dir()?.into_os_string();
+
+        if let Some(path) = current_dir_path.to_str() {
```

Comment:
instead of keeping the logs file in memory you could just push the new log line to the file ?